lib/mix/tasks/hex.repo.ex

```diff
@@ -196,6 +201,22 @@ defmodule Mix.Tasks.Hex.Repo do
     ssh_hostkey_fingerprint(public_key)
   end
 
+  defp fetch_public_key(nil, _, _), do: nil
+
+  defp fetch_public_key(fingerprint, repo_url, auth_key) do
+    case Hex.Repo.get_public_key(repo_url, auth_key) do
+      {:ok, {200, key, _}} ->
+        if show_public_key(key) == fingerprint do
+          key
+        else
+          Mix.raise("public key fingerprint does not match with provided")
+        end
+
+      {:error, _} ->
+        Mix.raise("could not download public key. check your auth key and the server")
```

Comment:
I think it's important to show the reason in error message. We have a helper for that:

```suggestion
      other ->
        Hex.Shell.error("Downloading public key failed")
        Hex.Utils.print_error_result(other)
        Mix.Tasks.Hex.set_exit_code(1)
```

@ericmj btw, when do we `Mix.raise` vs `Hex.Shell.error`? The latter obviously makes sense when we want to od more work after but often it is the last expression in a given task anyway.